### ISSUE TITLE
setNodeModels() strict requirement of parent nodes in model

### ISSUE THREAD

**Issue by jarrod-dalton** [nutterb/HydeNet#48]
Is there a downside to loosening the requirement that a model object must contain all parent nodes present in the graph?

```
net <- HydeNetwork(~ card1.ace | card1
                   + card2.ace | card2
                   + initialPoints | card1*card2*card1.ace*card2.ace
                   + hit1 | initialPoints*dealerUpcard
                   + card3 | hit1
                   + card3.ace | card3
                   + pointsAfterCard3 | initialPoints*card3*card3.ace
                   + hit2 | pointsAfterCard3*dealerUpcard
                   + card4 | hit2
                   + card4.ace | card4
                   + pointsAfterCard4 | pointsAfterCard3*card4*card4.ace
                   + hit3 | pointsAfterCard4*dealerUpcard
                   + card5 | hit3
                   + card5.ace | card5
                   + pointsAfterCard5 | pointsAfterCard4*card5*card5.ace
                   + playerFinalPoints | initialPoints*hit1*pointsAfterCard3
                                         *hit2*pointsAfterCard4*hit3*pointsAfterCard5
                   + dealerFinalPoints | dealerUpcard
                   + payoff | playerFinalPoints*dealerFinalPoints)

data(BlackJackTrain)

# glm.hit1 <- glm(hit1 ~ initialPoints + I(dealerUpcard %in% c("9","10-K","A")),
#                 data = BlackJackTrain, family="binomial")

#this model is missing 'dealerUpcard' even though it's in the graph:
glm.hit1 <- glm(hit1 ~ initialPoints, data = BlackJackTrain, family="binomial")

net <- setNodeModels(net, glm.hit1)#, glm.hit2, glm.hit3)
```

```
Error in setNodeModels(net, glm.hit1) : 
1: The following model independent variables are not identical to the node parent list: hit1
```

**Comment by nutterb** [comment]
If we relaxed this requirement, I'm not entirely sure what would happen down stream.  Initially, the `nodeFormula` element would be redefined, but the `parents` element would go unchanged.  That will definitely create a discrepancy between the actual model structure and the structure represented by the graph (the graphing relies mostly on the `nodes` and `parents` elements).  

I _think_ everything else would continue to work as expected.  The new `nodeFormula` would get passed to `writeJagsFormula`, proceed to `writeJagsModel`, and then get included in `writeNetworkModel`.  

That being said, I think we should make a decision on what we think is most appropriate for the user to do at this point.  In my mind, maintaining consistency between the plot and the actual structure is crucial.  The two options we have are
1. Assume that the model structure passed by the user is what is intended and print a warning that the parent relationship has been deleted.
2. Assume that the original model structure specified by the user is correct and the model is in error.  This would require the user to correct the initial model structure before proceeding.

Personally, I am partial to option 2.  If I'm honest, part of my preference is that it's already coded :).  but part of it is also that I have hesitations about letting people alter their model structure on the fly.  I would rather be forced to stop and reconsider why my original and actual model definitions don't match rather than get a working warning that I may not notice if I've buried it in the middle of a long script.

Your thoughts?
